File: Algorithms/SortingSeaching/BucketSort.py

def bucket_sort(nums, num_buckets=4):

    buckets = [ [] for _ in range(num_buckets) ]

    mx = max(nums) + 1

    for n in nums:

        index = int(n / mx * num_buckets)

        buckets[index].append(n)

    array_index = 0

    for bucket in buckets:

        bucket.sort()

        for num in bucket:

            nums[array_index] = num

            array_index += 1

def main():

    nums = [12, 32, 43, 11, 58, 60, 68, 90, 80, 99, 22]

    print(nums)
    bucket_sort(nums)
    print(nums)

if __name__ == '__main__':
    main()

File: Algorithms/SortingSeaching/BucketSort.js

bucketSort = (nums, numBuckets = 4) => {

    let buckets = [];
    while (buckets.push([]) < numBuckets);

    mx = Math.max(...nums) + 1

    for (let n of nums) {

        let index = Math.floor(n / mx * numBuckets);

        buckets[index].push(n);

    }

    let aryIndex = 0;

    for (let bucket of buckets) {

        bucket.sort((a, b) => a - b);

        for (let n of bucket) {

            nums[aryIndex++] = n;

        }

    }

}

let nums = [12, 32, 43, 11, 58, 60, 68, 90, 80, 99, 22];

console.log(nums);

bucketSort(nums);

console.log(nums);
